cfgov/jinja2/v1/_layouts/base.html

```diff
@@ -252,83 +257,66 @@
-      {% endif %}
-    }
+    {% macro template_js() %}
+       {% include 'js/routes/' + request.path.split('/')[1] + '/single.js' ignore missing %}
+    {% endmacro %}
+    {% set js_source = template_js() | trim %}
+    {% if js_source | length > 0 %}
+      a.push( '{{ static('js/routes/' + request.path.split('/')[1] + '/single.js') }}' );
+    {% endif %}
+
+    {#
+      Check and include component-level JavaScript.
+      Component-level JavaScript adds scripts appearing in
+      unprocessed/js/routes/on-demand/[component].js
+      to the page, based on what components were added in Wagtail.
+    #}
+    {% if page and page.media %}
+       {% for js in page.media %}
+         a.push( '{{ static('js/routes/on-demand/' + js) }}' );
+       {% endfor %}
+    {% endif %}
 
     {#
       Check and include page-level JavaScript.
       Page-level JavaScript applies to a single pages under the URL scheme
       www.consumerfinance.gov/[app-url]/somepage
       and JavaScript appears in unprocessed/js/routes/[app-url]/index.js.
     #}
-    function loadScriptPage(){
-      {# Check and include page-level JavaScript. #}
-      {% macro page_js() %}
-         {% include 'js/routes/' + request.path[1:] + 'index.js' ignore missing %}
-      {% endmacro %}
-      {% set js_source = page_js() | trim %}
-      {% if js_source | length > 0 %}
-        loadScript('{{ static('js/routes/' + request.path[1:] + 'index.js') }}');
-      {% endif %}
-    }
+    {% macro page_js() %}
+       {% include 'js/routes/' + request.path[1:] + 'index.js' ignore missing %}
+    {% endmacro %}
+    {% set js_source = page_js() | trim %}
+    {% if js_source | length > 0 %}
+      a.push( '{{ static('js/routes/' + request.path[1:] + 'index.js') }}' );
+    {% endif %}
 
-    {#
-      Modernizr adds a no-js class the HTML element if ECMAScript 5 features
-      are not available. Check for that CSS class here and act accordingly.
-    #}
-    if (document.body.parentElement.className.indexOf('no-js')===-1){
-      {# Include site-wide JavaScript. #}
-      loadScript('{{ static('js/routes/common.js') }}', function(){
-        loadScriptTemplate();loadScriptComponent();loadScriptPage();
-      });
-    }
-</script>
+    {# Minified dynamic JavaScript loader that injects a script tag in the head of the page #}
+    for(var b,c,d=[],f=document.scripts[0];b=a.shift();)c=document.createElement("script"),"async"in c?(c.async=!1,c.src=b,document.head.appendChild(c)):(d.push(c),c.onreadystatechange=function(){for(var e;d[0]&&"loaded"==d[0].readyState;)e=d.shift(),e.onreadystatechange=null,f.parentNode.insertBefore(e,f)},c.src=b);
+    }();
+  }
 
-<script>
-//<![CDATA[
-    var usasearch_config = { siteHandle: 'cfpb' };
-    loadScript('https://search.usa.gov/assets/sayt_loader.js');
-//]]>
+  var usasearch_config = { siteHandle: 'cfpb' };
 </script>
 
+<script async src="https://search.usa.gov/javascripts/remote.loader.js"></script>
```

Comment:
Do we need the change shown here https://github.com/cfpb/cfgov-refresh/pull/3791 ?

Reply:
Fixed.